h/views/account_signup.py

```diff
@@ -53,24 +54,24 @@ def post(self):
             return {"form": self.form.render()}
 
         signup_service = self.request.find_service(name="user_signup")
-        signup_service.signup(
-            username=appstruct["username"],
-            email=appstruct["email"],
-            password=appstruct["password"],
-            privacy_accepted=datetime.datetime.utcnow(),
-        )
-
-        self.request.session.flash(
-            jinja2.Markup(
-                _(
-                    "Please check your email and open the link to activate your "
-                    "account."
-                )
-            ),
-            "success",
-        )
-
-        return httpexceptions.HTTPFound(location=self.request.route_url("index"))
+        try:
+            signup_service.signup(
+                username=appstruct["username"],
+                email=appstruct["email"],
+                password=appstruct["password"],
+                privacy_accepted=datetime.datetime.utcnow(),
+            )
+            greeting = "Account registration successful"
+            message = _(
+                "Please check your email and open the link to activate your " "account."
+            )
+        except ConflictError as e:
+            greeting = "Account already registered"
+            message = _(
+                "{failure_reason} Please check your email and open the link to activate your "
+                "account.".format(failure_reason=e.message)
+            )
+        return {"greeting": greeting, "message": message}
```

Comment:
The meaning of "greeting" wasn't clear to me initially. Perhaps call this "title" or "heading" instead?